### ISSUE TITLE
Critical bug: Consumer credentials don't work after change

### ISSUE THREAD

**Issue by natt-tester** [GluuFederation/gluu-gateway#159]
Logged in to GG, a Consumer created with OAuth credentials
Steps:
1. Change all OAuth credentials to different strings and save them
2. Make a get-token call with the new credentials
3. Make a get-token call with the old credentials

Expected: After the credentials update I expect them to work instead of the old ones. The old ones don't work.
Actual: The call with the new creds generates an internal error. The old credentials work and the call generates a token.

![2018-06-20_16h47_53](https://user-images.githubusercontent.com/38302725/41642254-23b85f2e-74ab-11e8-8afc-7d289ffed010.png)

**Comment by ldeveloperl1985** [comment]
@natt-tester I've solved this issue. Please check

**Comment by natt-tester** [comment]
@ldeveloperl1985, now I can't create Consumer credentials at all... Watch the video -- tested on dev1 using Firefox:

[2018-06-21_16h43_16.zip](https://github.com/GluuFederation/gluu-gateway/files/2122456/2018-06-21_16h43_16.zip)

**Comment by ldeveloperl1985** [comment]
I see on dev1. There is an old code. Please check with the latest code.

**Comment by natt-tester** [comment]
The issue is blocked by #160.

**Comment by natt-tester** [comment]
It works better with the new code. I tried to change the credentials to random strings and it didn't work:
![2018-06-21_19h13_31](https://user-images.githubusercontent.com/38302725/41709721-420eb89c-7587-11e8-9d32-b6d0b14d6687.png)
The only change I managed to make was by:

1. creating other credentials
2. copying them to a notepad
3. removing them from the list
4. copying the credentials from the notepad to an already existing card by removing the original ones. 

Then I made a get-client-token request and it worked, but when I tried to use the original credentials from step 4, it gave me an internal error, so I'm not sure if we have everything covered:

![2018-06-21_19h20_14](https://user-images.githubusercontent.com/38302725/41710115-3b4d1674-7588-11e8-98d2-c9742b8ab615.png)
